catalog/services_state.go

```diff
@@ -147,6 +148,9 @@ func (state *ServicesState) HasServer(hostname string) bool {
 
 // A server has left the cluster, so tombstone all of its records
 func (state *ServicesState) ExpireServer(hostname string) {
+	state.Lock()
+	defer state.Unlock()
```

Comment:
I moved the locks from services_delegate.go here, given that's the only place where this function is called.